### PR TITLE
fix: docs for js library #458

### PR DESCRIPTION
**Pull request by abubakarsohail**
fix #458

This PR adds relevant docs  and code snippets for JavaScript library.

### CODE REVIEW

File: features/prompt-history/request-id.mdx [hunk 1]
```diff
@@ -33,6 +33,23 @@ response, pl_request_id = openai.Completion.create(
 )
 ```
 
+## Javascript
+
+To retrieve the `pl_request_id` using OpenAI in JavaScript (through `promptlayer.OpenAI`), set the argument `return_pl_id` to `true` in your function call. This will change the call to now return the OpenAI response and the `pl_request_id`. If you are using `stream: true`, then only the last `pl_request_id` will be the id; otherwise, it will be `undefined`.
+
+The same is true for Anthropic.
+        
+```js
```
Comment (jzone3) [inline features/prompt-history/request-id.mdx:42]: Should we add a Typescript example here as well? Just for that ignore line?

File: features/prompt-registry.mdx [hunk 2]
```diff
@@ -258,13 +368,67 @@ response = openai.ChatCompletion.create(
...
+  prompt_template: {
+    function_call: "auto", // or {"name": function_name} if you want to make sure the function gets called.
+    input_variables: [], // list them here if it has any
+    messages: messages.map(({ role, content }) => ({
+      role,
+      prompt: {
+        input_variables: [], // list them here if it has any
+        template: content,
+        template_format: "f-string", // or "jinja"
+        validate_template: true, // optional
```
Comment (jzone3) [inline features/prompt-registry.mdx:404]: What is validate_template??

File: features/prompt-registry.mdx [hunk 3]
```diff
@@ -258,13 +368,67 @@ response = openai.ChatCompletion.create(
...
+        _type: "prompt",
+      },
+    })),
+    functions,
+    _type: "chat_promptlayer_langchain", // important if using gpt-3.5-turbo or gpt-4
```
Comment (jzone3) [inline features/prompt-registry.mdx:409]: This is crazy... do we really need this?

**Review by jzone3 — CHANGES_REQUESTED** [review]
Great job. Just a few comments and message on slack about underscores vs camelCase

File: features/prompt-history/request-id.mdx (hunk 1 above)
Comment (abubakarsohail) [reply features/prompt-history/request-id.mdx:42]: @jzone3 there is a code snippet [here](https://github.com/MagnivOrg/prompt-layer-docs/blob/9b7f09dd4ea4709ebe3db3d25133dd2f90484c14/languages/javascript.mdx#L64) for TypeScript. I think that is sufficient enough.

File: features/prompt-registry.mdx (hunk 2 above)
Comment (abubakarsohail) [reply features/prompt-registry.mdx:404]: @jzone3 this is coming from langchain. Should we remove this? We are eventually going to remove our dependency on langchain.

File: features/prompt-registry.mdx (hunk 3 above)
Comment (abubakarsohail) [reply features/prompt-registry.mdx:409]: @jzone3 this is how the backend accepts the prompt template to be.

File: features/prompt-registry.mdx (hunk 2 above)
Comment (jzone3) [reply features/prompt-registry.mdx:404]: I hate it but good point lol

**Review by jzone3 — APPROVED** [review]

**Pull request merged**